src/pixelator/pna/read_processing/pipeline.py

```diff
@@ -64,27 +66,44 @@ class AmpliconPipeline(Pipeline):
     :param post_steps: A list of steps that are applied to the reads after the combining step.
     """
 
-    paired = True
+    paired: bool | None = None
 
     def __init__(
         self,
         combiner: CombiningModifier,
-        pre_modifiers: Iterable[PairedEndModifier] | None = None,
-        pre_steps: Iterable[PairedEndStep] | None = None,
+        pre_modifiers: (
+            Iterable[
+                Union[
+                    PairedEndModifier,
+                    SingleEndModifier,
+                    tuple[SingleEndModifier | None, SingleEndModifier | None],
+                ]
+            ]
```

Comment:
I'm wondering if perhaps this type annotation can be simplified into `PairedEndModifier | SingleEndModifier | tuple[SingleEndModifier | None, SingleEndModifier | None]`? I think `Union` and `|` do the same thing, so mixing them makes it a bit more difficult t read do me.